pylint/pyreverse/writer.py

```diff
@@ -159,53 +179,33 @@ def get_values(self, obj):
 
                 label = fr"{label}{func.name}({args}){return_type}\l"
             label = "{%s}" % label
-        if is_exception(obj.node):
-            return dict(fontcolor="red", label=label, shape="record")
-        return dict(label=label, shape="record")
-
-    def close_graph(self):
-        """print the dot graph into <file_name>"""
-        self.printer.generate(self.file_name)
+        properties = NodeProperties(
+            label=label,
+            fontcolor="red" if is_exception(obj.node) else "black",
+            color="black",
+        )
+        return properties
 
 
 class VCGWriter(DiagramWriter):
     """write vcg graphs from a diagram definition and a project"""
 
-    def __init__(self, config):
-        styles = [
-            dict(arrowstyle="solid", backarrowstyle="none", backarrowsize=0),
-            dict(arrowstyle="solid", backarrowstyle="none", backarrowsize=10),
-            dict(
-                arrowstyle="solid",
-                backarrowstyle="none",
-                linestyle="dotted",
-                backarrowsize=10,
-            ),
-            dict(arrowstyle="solid", backarrowstyle="none", textcolor="green"),
-        ]
-        DiagramWriter.__init__(self, config, styles)
-
```

Comment:
The ``__init__`` is now obsolete. It formerly defined the styles to apply to the different types of edges for displaying relationships.
These are now defined in the ``ARROWS`` dictionary in the module of the corresponding printer class.